src/coreclr/jit/compiler.cpp

```diff
@@ -5134,6 +5134,29 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     StackLevelSetter stackLevelSetter(this);
     stackLevelSetter.Run();
 
+    // Block layout should not change at this point.
+    // Do one last pass to reverse any conditions that might yield more fall-through branches.
+    if (opts.OptimizationEnabled())
+    {
+        for (BasicBlock* const block : Blocks())
+        {
+            if (block->KindIs(BBJ_COND) && block->CanRemoveJumpToTarget(block->GetTrueTarget(), this))
+            {
+                // Reverse the jump condition
+                GenTree* test = block->lastNode();
+                assert(test->OperIsConditionalJump());
+                GenTree* cond = gtReverseCond(test);
+                assert(cond == test); // Ensure `gtReverseCond` did not create a new node.
+
+                BasicBlock* newFalseTarget = block->GetTrueTarget();
+                BasicBlock* newTrueTarget  = block->GetFalseTarget();
+                block->SetTrueTarget(newTrueTarget);
+                block->SetFalseTarget(newFalseTarget);
+                assert(block->CanRemoveJumpToTarget(newFalseTarget, this));
+            }
+        }
+    }
```

Comment:
This should be a proper phase ensuring post-phase checks run, and ensuring the changed IR nodes are dumped.

Reply:
I was planning on removing this change from this PR, and possibly introducing it if/when we remove the condition reversals done in earlier phases. I'll add it as a new phase then.

Reply:
No worries! I appreciate the feedback -- this should be ready soon.

Reply:
> this should be ready soon.

Famous last words... closing this in favor of #97488.